src/components/BulkOperations/index.jsx

```diff
@@ -0,0 +1,208 @@
+import { useEffect, useState } from 'react';
+import {Link} from 'react-router-dom'
+import classNames from 'classnames';
+import Papa from 'papaparse';
+import {
+  SelectMenu, MenuItem, Dropzone, DataTable, Alert,
+} from '@openedx/paragon';
+
+import Collapsible from '../Collapsible';
+import LoadingSpinner from '../LoadingSpinner';
+import { formatDate } from '../../utils';
+import DiscoveryDataApiService from '../../data/services/DiscoveryDataApiService';
+import './BulkOperations.scss';
+
+const availableOperations = {
+  course_create: 'Bulk Create',
+  course_partial_update: 'Bulk Course Update',
+  course_run_partial_update: 'Bulk CourseRun Update',
+  course_rerun: 'Bulk Rerun',
+};
+
+const BulkOperations = () => {
+  const [bulkOperationId, setBulkOperationId] = useState(null);
+  const [fileContent, setFileContent] = useState(null);
+  const [file, setFile] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [submitSuccess, setSubmitSuccess] = useState(null);
+  const [historicalRecords, setHistoricalRecords] = useState(null);
```

Comment:
nit: historicalBulkOperations or historicalBulkRecords or similar.

Reply:
historicalRecords seems to convey the context just fine imo